lib/srv/authhandlers.go

```diff
@@ -332,8 +331,11 @@ func (h *AuthHandlers) HostKeyAuth(addr string, remote net.Addr, key ssh.PublicK
 // hostKeyCallback allows connections to hosts that present keys only if
 // strict host key checking is disabled.
 func (h *AuthHandlers) hostKeyCallback(hostname string, remote net.Addr, key ssh.PublicKey) error {
+	// Use the server's shutdown context.
```

Comment:
Naïve question: Why is this context a good choice here? Is the only cancellation case when the server wants to shut down and has an outstanding task?

Reply:
I'm not completely sure on this one but here was my thought. This function implements `ssh.HostCallback` and is used by the ssh package, so it isn't feasible to add the context in ourselves, leaving the server context as the only available option which is better than no context. context.TODO should be used in places where we expect to add it at some point but it's just out of scope of the current work.

Reply:
Sounds reasonable.